### ISSUE TITLE
Example Site Scripts Invalid Against Latest Site Script Schema

### ISSUE THREAD

**Issue by joselarios** [SharePoint/sp-dev-docs#5361]
## Category

- [x] Bug

## Describe the bug

Some Site Script examples in the latest documentation for [Site design JSON schema](https://docs.microsoft.com/en-us/sharepoint/dev/declarative-customization/site-design-json-schema) fail to validate against the [latest Site Script Schema](https://developer.microsoft.com/json-schemas/sp/site-design-script-actions.schema.json). 

Certain Site Script examples do validate (with some modifications to the JSON site script as noted in "**N.B.**" below). I can provide examples of those if needed.

I might have tracked the issue to the way the "additionalProperties" property is specified in several action definitions, but I need confirmation in case the validation error lies elsewhere.

## Steps to reproduce

1. Use the example JSON in the [Define a new content type](https://docs.microsoft.com/en-us/sharepoint/dev/declarative-customization/site-design-json-schema#define-a-new-content-type) section of the documentation to form a site script. For reference, the following is a site script constructed in this way:

`{
  "$schema": "schema.json",
  "actions": [
    {
      "verb": "createContentType",
      "name": "Contoso Projects",
      "description": "custom list content type",
      "parentName": "Item",
      "hidden": false,
      "subactions": [
        {
          "verb": "addSiteColumn",
          "internalName": "siteColumn1Text"
        },
        {
          "verb": "addSiteColumn",
          "internalName": "siteColumn2Number"
        },
        {
          "verb": "addSiteColumn",
          "internalName": "siteColumn3Note"
        }
      ]
    }
  ]
}`

**N.B.** While the documentation specifies it, I have removed the following properties from this Site Script: "bindingdata" and "version". When these properties are included, the Site Script JSON will always fail to validate against the latest Site Script Schema, because there is an "additionalProperties" property that prohibits them—please refer to the [Site Script Schema](https://developer.microsoft.com/json-schemas/sp/site-design-script-actions.schema.json) for confirmation. In the previous (draft-06) Site Script Schema, those properties were defined and allowed.

2. Use a JSON Schema validator to validate the JSON above against the latest [Site Script Schema](https://developer.microsoft.com/json-schemas/sp/site-design-script-actions.schema.json). E.g. I have used [JSON Schema Validator](https://www.jsonschemavalidator.net/) as well as [Ajv](https://ajv.js.org/) to confirm the invalidity.

3. See validation error.

## Expected behavior

JSON Site Script example should be valid against latest Site Script Schema.

### Additional context

The main validation issue seems to be that the Site Script action has properties the schema does not allow. However, the errors seem to suggest that none of the properties are allowed by the Site Script Schema, even the required ones for this example (e.g. verb, name, hidden).

This suggests that the error is in how the "additionalProperties" property is set in the Site Script Schema for the action. I can provide additional information on this if needed.

**Comment by bcameron1231** [comment]
Yup. So bindata and version was removed from the latest schema. I don't think this is a bug per se, but I think all we need to do is update the documentation page to remove the old schema declarations to match the new schema.

**Comment by joselarios** [comment]
Hi @bcameron1231,

I think there may be two related issues here:

(1) As you mentioned, the documentation needs to be updated to exclude "bindata" and "version", as any Site Script with those properties will be invalid against the latest Site Script Schema.

(2) The Site Script Schema itself is incorrectly formed with respect to action schemas that have both the "additionalProperties" and "allOf" property—the "additionalProperties" is nested incorrectly in the Schema, or so I suspect. This is the larger problem which I outlined above, and which would require updating the Site Script Schema itself.

**Comment by pschaeflein** [comment]
I can confirm point [(2) above](https://github.com/SharePoint/sp-dev-docs/issues/5361#issuecomment-589859872). Removing the "additionalProperties" setting from definitions that have "allOf" is the fix for many validation errors. (That is also the approach I've taken in [my VSCode extension](https://marketplace.visualstudio.com/items?itemName=paulschaeflein.sitedesign-schema).)
